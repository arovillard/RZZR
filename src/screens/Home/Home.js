import { useTheme, useNavigation } from '@react-navigation/native';
import React from 'react';
import { Text, View } from 'react-native';
import { Config } from 'react-native-config';
import { useSelector } from 'react-redux';
import { strings } from '@/localization';
import { getUser } from '@/selectors/UserSelectors';
import { styles } from '@/screens/Home/Home.styles';
import { typography } from '@/theme';
import { Button } from '@/components';

export function Home() {
  const { colors } = useTheme();
  const user = useSelector(getUser);
  const navigation = useNavigation();

  return (
    <View style={styles.container}>
      <Text style={[typography.title, { color: colors.text }]}>
        {strings.home.message} {user?.username}
      </Text>
      <Text style={[typography.text, { color: colors.text }]}>
        {strings.home.variant} {Config.BUILD_VARIANT}
      </Text>
      <Button
        onPress={() => navigation.navigate('CustomerNavigator', { screen: 'Customer' })}
        style={styles.submitButton}
        title="View customer profile ci test"
      />
    </View>
  );
}
