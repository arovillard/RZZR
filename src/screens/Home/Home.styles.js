import { StyleSheet } from 'react-native';
import { spacing } from '@/theme';

export const styles = StyleSheet.create({
  container: {
    paddingTop: 50,
    flex: 1,
  },
  headerText: {
    paddingHorizontal: spacing.xs,
  },
  subContainer: {
    flex: 1,
    backgroundColor: 'white',
  },
  submitButton: {
    marginTop: spacing.m,
    marginBottom: spacing.m,
  },
  pageHeader: {
    marginBottom: spacing.xs,
  },
  pageSubheader: {
    marginBottom: spacing.s,
  },
});
