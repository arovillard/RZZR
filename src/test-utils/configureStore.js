import { applyMiddleware, legacy_createStore as createStore } from 'redux';
import thunk from 'redux-thunk';
import { rootReducer } from '@/reducers';
import { initialStore } from '@/store';

export function configureStore({
  initialState = initialStore,
  demoMode = false,
  networkService = {},
}) {
  return createStore(
    rootReducer,
    initialState,
    applyMiddleware(thunk.withExtraArgument({ networkService, demoMode }))
  );
}
