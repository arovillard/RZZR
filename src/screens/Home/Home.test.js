import { render } from '@testing-library/react-native';
import React from 'react';
import { Home } from '@/screens/Home/Home';
import { withProviders } from '@/test-utils';

const fakeStore = {
  error: {},
  status: {},
  user: {
    name: 'johndoe',
  },
  customer: {
    customers: [
      {
        email: 'example@test02.com',
        id: 10076,
        is_active: true,
        is_person: false,
        name: 'RM test 02',
        phone: '2222222',
      },
      {
        email: 'example@test01.com',
        id: 10075,
        is_active: true,
        is_person: false,
        name: 'RM test 01',
        phone: '1231231',
      },
    ],
  },
};

describe('Home', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(withProviders(<Home />, { initialState: fakeStore }));

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render a welcome message with the user name', () => {
    const { getByText } = render(withProviders(<Home />, { initialState: fakeStore }));

    expect(getByText(`Hello,`)).toBeTruthy();
    expect(getByText(`johndoe`)).toBeTruthy();
  });
});
