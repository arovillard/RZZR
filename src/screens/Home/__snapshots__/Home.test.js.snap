// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 40,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 18,
          "fontWeight": "700",
        },
        Object {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
  >
    Welcome
     
    johndoe
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
          "fontWeight": "400",
        },
        Object {
          "color": "rgb(28, 28, 30)",
        },
      ]
    }
  >
    App variant: 
     
    TEST
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "rgb(216, 216, 216)",
        "borderRadius": 5,
        "borderWidth": 1,
        "justifyContent": "center",
        "marginTop": 24,
        "opacity": 1,
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgb(28, 28, 30)",
          },
          Object {
            "fontSize": 16,
            "fontWeight": "700",
          },
          null,
        ]
      }
    >
      View customer profile
    </Text>
  </View>
</View>
`;
